index.js

```diff
@@ -10,8 +10,49 @@ var resolveRc = require('./lib/resolve-rc.js');
 var pkg = require('./package.json');
 var path = require('path');
 
+/**
+ * Error thrown by Babel formatted to conform to Webpack reporting.
+ */
+function BabelLoaderError(name, message, codeFrame, hideStack, error) {
+  Error.call(this);
+  Error.captureStackTrace(this, BabelLoaderError);
+
+  this.name = 'BabelLoaderError';
+  this.message = formatMessage(name, message, codeFrame);
+  this.hideStack = hideStack;
+  this.error = error;
+}
+
+BabelLoaderError.prototype = Object.create(Error.prototype);
+
+var STRIP_FILENAME_RE = /^[^:]+: /;
+
+var formatMessage = function(name, message, codeFrame) {
+  return (name ? name + ': ' : '') + message + '\n\n' + codeFrame + '\n';
+};
+
 var transpile = function(source, options) {
-  var result = babel.transform(source, options);
+  var result;
+  try {
+    result = babel.transform(source, options);
+  } catch (error) {
+    if (error.message && error.codeFrame) {
+      var message = error.message;
+      var name;
+      var hideStack;
```

Comment:
@hzoo the option is [debug](https://webpack.github.io/docs/loaders.html#debug), and one can use it like so: https://github.com/nkt/yml-loader/blob/e152f78b06a9580a4ac827588aafa9a7b8d1e316/index.js#L34, which means it is also overridable by an individual loader config via a query.